### PR TITLE
Added a regression test for #25024

### PR DESCRIPTION
**Pull request by VSadov**
Test-only change.

Fixes:https://github.com/dotnet/roslyn/issues/25024

### CODE REVIEW

**Comment by VSadov** [comment]
@jcouv - please take a look. This is a test-only change.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs [hunk 1]
```diff
@@ -3590,5 +3590,95 @@ static void Main()
...
+                local2 = 2;
+                System.Console.Write(local1 + local2);
+            }
+
+            if (flag)
+            {
+                ref var local1 = ref M1(out var _);
+                ref var local2 = ref M1(out var _);
+
+                local1 = 3;
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs]: I didn't understand the purpose of the second block.

Comment (VSadov) [reply src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs]: In theory we could reuse "leaked" temps from different blocks, but we don't (for simplicity).
I have added a second block just to see whether such reuse happens or not.

**Comment by VSadov** [comment]
@dotnet-bot test windows_debug_unit64_prtest  please

**Pull request merged**